src/actors/objects/paths.rs

```diff
@@ -2,7 +2,56 @@ use std::fmt::Write;
 
 use symbolic::common::Uuid;
 
-use crate::types::{DirectoryLayout, DirectoryLayoutType, FileType, FilenameCasing, ObjectId};
+use crate::actors::objects::DownloadPath;
+use crate::types::{
+    DirectoryLayout, DirectoryLayoutType, FileType, FilenameCasing, ObjectId, SourceFilters,
+};
+
+/// Yield download paths given parameters from the source config
+pub fn prepare_download_paths<'a>(
+    filetypes: impl Iterator<Item = FileType> + 'a,
+    filters: &'a SourceFilters,
+    layout: DirectoryLayout,
+    object_id: &'a ObjectId,
+) -> impl Iterator<Item = DownloadPath> + 'a {
+    filetypes.filter_map(move |filetype| {
+        if (filters.filetypes.is_empty() || filters.filetypes.contains(&filetype))
+            && matches_path_prefixes(object_id, &filters.path_prefixes)
+        {
+            Some(DownloadPath(get_directory_path(
+                layout, filetype, &object_id,
+            )?))
+        } else {
+            None
+        }
+    })
+}
+
+fn matches_path_prefixes(object_id: &ObjectId, prefixes: &[String]) -> bool {
```

Comment:
If we choose to keep this, let's add a utility function to `symbolic_common` that does this. It fits quite well alongside the other stuff in `symbolic/common/src/path.rs`.

Reply:
To me this is just an aspect of the symbolicator protocol. The heavy lifting is done by `glob` anyway.